AutoCollection/Statsbeat.ts

```diff
@@ -209,42 +201,64 @@ class Statsbeat {
+    private _getNetworkStatsbeatCounter(category: number, endpoint: string): Network.NetworkStatsbeat {
+        // Check if counter is available
+        for (let i = 0; i < this._networkStatsbeatCollection.length; i++) {
+            // Same object
+            if (category === this._networkStatsbeatCollection[i].category &&
+                endpoint === this._networkStatsbeatCollection[i].endpoint) {
+                return this._networkStatsbeatCollection[i];
+            }
         }
-        this._lastRequests = requests;
+        // Create a new one if not found
+        let newCounter = new Network.NetworkStatsbeat(category, endpoint);
+        this._networkStatsbeatCollection.push(newCounter);
+        return newCounter;
     }
 
-    private _trackRequestsCount(properties: {}) {
-        if (this._totalSuccesfulRequestCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_SUCCESS, value: this._totalSuccesfulRequestCount, properties: properties });
-            this._totalSuccesfulRequestCount = 0; //Reset
-        }
-        if (this._totalFailedRequestCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_FAILURE, value: this._totalFailedRequestCount, properties: properties });
-            this._totalFailedRequestCount = 0; //Reset
-        }
-        if (this._retryCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.RETRY_COUNT, value: this._retryCount, properties: properties });
-            this._retryCount = 0; //Reset
-        }
-        if (this._throttleCount > 0) {
-            this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.THROTTLE_COUNT, value: this._throttleCount, properties: properties });
-            this._throttleCount = 0; //Reset
+    private _trackRequestDuration(commonProperties: {}) {
+        for (let i = 0; i < this._networkStatsbeatCollection.length; i++) {
+            var currentCounter = this._networkStatsbeatCollection[i];
+            currentCounter.time = +new Date;
+            var intervalRequests = (currentCounter.totalRequestCount - currentCounter.lastRequestCount) || 0;
+            var elapsedMs = currentCounter.time - currentCounter.lastTime;
+            var averageRequestExecutionTime = ((currentCounter.intervalRequestExecutionTime - currentCounter.lastIntervalRequestExecutionTime) / intervalRequests) || 0;
+            currentCounter.lastIntervalRequestExecutionTime = currentCounter.intervalRequestExecutionTime; // reset
+            if (elapsedMs > 0 && intervalRequests > 0) {
+                // Add extra properties
+                let properties = Object.assign({ "category": this._networkStatsbeatCollection[i].category, "endpoint": this._networkStatsbeatCollection[i].endpoint }, commonProperties);
+                this._statbeatMetrics.push({ name: Constants.StatsbeatCounter.REQUEST_DURATION, value: averageRequestExecutionTime, properties: properties });
```

Comment:
I think part of the specs says that if the value of the stats is 0 for network stats, we don't send the metric? Might need a check before appending to `_statsbeatMetrics`

Reply:
Check must be right there a few lines before, we don't send duration unless there was a request and elapsed milliseconds